lib/engine/game/g_18_india/step/railhead_tracker.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/tracker'
+
+module Engine
+  module Game
+    module G18India
+      module Step
+        module RailheadTracker
+          # Implements track laying rule for multiple yellow tiles
+          # Track laying starts at a railhead(token) and must extend a route(path) from all laid tiles
+
+          def setup
+            super
+            @round.next_empty_hexes = nil
+            @round.laid_yellow_hexes = []
+          end
+
+          def round_state
+            super.merge({ next_empty_hexes: nil, laid_yellow_hexes: [] })
+          end
+
+          def calculate_railhead_hexes
+            return [] if @round.laid_yellow_hexes.empty?
+
+            # check simple case of only one 'white' neighbor connected to prior tile => return without walking
+            last_yellow_hex = @round.laid_yellow_hexes.last
+            neighbors = last_yellow_hex.tile.exits.map { |e| last_yellow_hex.neighbors[e] }
+            empty_neighbors = neighbors.select { |h| h.tile.color == 'white' }
+            LOGGER.debug "railhead_hexes >> empty_neighbors: #{empty_neighbors.uniq.inspect}" \
+                         " exits: #{last_yellow_hex.tile.exits.inspect}"
+            return empty_neighbors if empty_neighbors.one? && last_yellow_hex.tile.exits.size <= 2 # exclude triple town tiles
+
+            corp = @round.current_operator
+            railheads = corp.placed_tokens.map(&:city)
+            placed_paths = @round.laid_yellow_hexes.map(&:tile).map(&:paths) # paths on all previously laid yellow hexes
+            LOGGER.debug "railhead_hexes >> corp: #{corp.inspect}, railheads: #{railheads.inspect}"
+            return [] unless railheads.any?
```

Comment:
It has failied in the case of a yellow OO tile, because the token got removed to be placed on one of the two cities.  But the "Simple Check" I added prior to walking eliminates the need for the railhead check.  Or so I think...  Will test a bit to confirm.

Reply:
Heh, I need to remove OO tiles from the simple check, as if the placing corp decides to put himself in the "unconnected" city, he has no path, even though there is an empty neighbor.

Reply:
![image](https://github.com/tobymao/18xx/assets/118568589/8415e464-e808-4b92-9eef-b407b97aeea3)
Railheads can be empty, but if empty, the walk simply doesn't happen.

Reply:
returning `nil` if the railheads are empty will allow method to be re-run after the OO token is placed.

Reply:
If I've understood this: the company only has its home token on the map, placed when the company floated. The home hex has just had a yellow OO tile placed on it, and the company's token has been removed and added to `pending_tokens`. Right?

It would be best to be checking `pending_tokens` before you get to this code path. The token location needs to be resolved before any more track can be laid.

Reply:
That is the correct situation where the railheads may be empty.  I have added the pending token check.

Reply:
> I was thinking that the `pending_token` check should be somewhere further back in the call stack. If there's a token to be placed then you shouldn't be in the track step at all.

That is tricky because I'm pretty sure this case is called by by the UI, not the steps.  The UI will call `available_hex` fairly often.